### DISCUSSION TITLE
Monitor the agent

### DISCUSSION THREAD

**Discussion by lecoqlibre** [Wimmics/solid-indexer#3] [Ideas]
How do we know that the agent is currently running?

We can write the agent status in a document on a POD but the agent can die without update its status.

One way of achieving this is to use a Web-socket between the client and the agent. Doing so, as long as the connection lives, the client can consider that the agent is running.

**Comment by FabienGandon** [comment]
in a hMAS ecosystem you could imagine several approaches : using the inbox of the pody of the agent to send a message/ping, rely on a service in the environment/workspace to get the list of running agents, etc.
